compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -181,6 +180,12 @@ pub(crate) fn type_check<'a, 'tcx>(
     let opaque_type_values =
         opaque_types::take_opaques_and_register_member_constraints(&mut typeck);
 
+    if let Some(polonius_context) = typeck.polonius_context.as_mut() {
+        let num_regions = infcx.num_region_vars();
+        let points_per_live_region = typeck.constraints.liveness_constraints.points();
+        polonius_context.record_live_regions_per_point(num_regions, points_per_live_region);
```

Comment:
I wonder instead of using an `Option` in PoloniusContext, if it's better to just use a different struct for MirTypeckResults...

Reply:
Yes, however I'd like to do something similar in a follow-up if you don't mind: what we eventually want to do I think is compute the localized constraints earlier (as they should only depend on mir typeck + liveness), and only store loan liveness in the context instead, with the optional debugging data that the polonius mir dump needs.